### PR TITLE
Add `upload_endpoint` plugin for FileSystem storage

### LINKED ISSUE #114: Update `PreparePresignImage` interactor for work with local filesystem
Now after switching to local filesystem storage when client call `presignData` mutation the error is raised:
```
undefined method `presign' for #<Shrine::Storage::FileSystem:0x000055a441021d78>
```
We should mount `upload_endpoint` module for this type of storage and return this URL with `presignData` mutation

### PR DESCRIPTION
**Pull request by EvgeniyEsaulkov**
### How it works
**Presign Data**
Screen Shot 2021-02-07 at 15 56 19

**Upload File**
Screen Shot 2021-02-07 at 15 56 48

**Update User**
Screen Shot 2021-02-07 at 15 57 21

### Test plan

List of steps to manually test introduced functionality:

* be sure that your app uses local storage:
```ruby
irb(main):001:0> Shrine.storages[:cache]
=> #<LocalStorage:0x000055f014900ab0 @_routes=nil, @prefix=#<Pathname:uploads/cache>, @directory=#<Pathname:/app/public/uploads/cache>, @permissions=420, @directory_permissions=493, @clean=true>
```
* Open GraphiQL App
* Make `presignData` request using schema:
```
mutation {
  presignData(
    filename: "test.png",
    type: "image/png"
  ) {
    url
    fields {
      key
      value
    }
  }
}
```
* Open Postman/API Tester or another tool to send request to upload endpoint (URL will be returned by `presignData` mutation)
* Send POST multipart/form-data request. Add fields returned by `presignData` mutation as fields for this request. Add selected image as `file` field
* Make `updateUser` request to attach uploaded image to the database record:
```
mutation($avatar: ImageUploader) {
  updateUser(
    avatar: $avatar
  ) {
    id
    avatarUrl
  }
}

Query variables:
{
  "avatar": {
    "id": <key-field-from-presign-data-response>,
    "storage": "cache",
    "metadata": {
      "filename": "test.png",
      "size":38720,
      "mimeType": "image/png"
    }
  }
}
```
### Review notes

While reviewing pull-request (especially when it's your pull-request),
please make sure that:

- you understand what problem is solved by PR and how is it solved
- new tests are in place, no redundant tests
- DB schema changes reflect new migrations
- newly introduced DB fields have indexes and constraints
- there are no missed files (migrations, view templates)
- required ENV variables added and described in `.env.example` and added to Heroku
- associated Heroku review app works correctly with introduced changes

### Deploy notes

- Add `HOST` ENV variable before deploying

### References
* Resolves #114

### CODE REVIEW

File: app/controllers/uploads_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class UploadsController < ApplicationController
+  def image
+    set_rack_response ImageUploader.upload_response(:cache, request.env)
+  end
+
+  private
+
+  def set_rack_response((status, headers, body))
```
Comment (ArthurZaharov) [inline app/controllers/uploads_controller.rb:8]: Maybe we can omit this method

File: app/storages/local_storage.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+class LocalStorage < Shrine::Storage::FileSystem
+  include Rails.application.routes.url_helpers
```
Comment (ArthurZaharov) [inline app/storages/local_storage.rb:2]: Could we replace it with delegate?

File: config/initializers/shrine.rb [hunk 3]
```diff
@@ -46,3 +46,6 @@ def local_storages
 end
 
 Shrine.storages = s3_options.values.all?(&:present?) ? amazon_s3_storages : local_storages
```
Comment (ArthurZaharov) [inline config/initializers/shrine.rb:48]:
```suggestion
if s3_options.values.all?(&:present?)
  Shrine.storages = amazon_s3_storages
else
  Shrine.plugin :upload_endpoint, upload_context: ->(request) { { key: request.params["key"] } }
  Shrine.storages = local_storages
end

```

File: app/controllers/uploads_controller.rb (hunk 1 above)
Comment (EvgeniyEsaulkov) [reply app/controllers/uploads_controller.rb:8]: Yes, we can. In this case the result will be less informative - just 204 status. But it is the same as we get when we're trying to upload file to AWS S3
Screen Shot 2021-02-13 at 10 33 00

**Review by ArthurZaharov — DISMISSED** [review]

**Comment by EvgeniyEsaulkov** [comment]
After testing with `react-next-base` app I added storage key to presign data - now it looks like 'cache/some-letters-and-digits.png' as we have by using AWS S3 storage

Also `LocalStorage#url` method added to display image properly: for `react-next-base` our app is remote, so `user.avatar_url` should be like "http://localhost:5050/public/uploads/avatar.png", not "public/uploads/avatar.png"

**Review by ArthurZaharov — APPROVED** [review]

**Pull request merged**